Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\EventListener\Linkvalidator;
+
+use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+final class BeforeRecordIsAnalyzedEventListener
+{
+    protected BrokenLinkRepository $brokenLinkRepository;
+    protected SoftReferenceParserFactory $softReferenceParserFactory;
+    public function __construct(
+        BrokenLinkRepository $brokenLinkRepository,
+        SoftReferenceParserFactory $softReferenceParserFactory
+    ) {
+        $this->brokenLinkRepository = $brokenLinkRepository;
+        $this->softReferenceParserFactory = $softReferenceParserFactory;
+    }
+
+    private function addItemToBrokenLinkRepository(string $table, string $field, array $newsItem, string $foundUrl, string $forbiddenDomain): void
+    {
+        $link = [
+            'record_uid' => $newsItem['uid'],
+            'record_pid' => $newsItem['pid'],
+            'language' => $newsItem['sys_language_uid'],
+            'field' => $field,
+            'table_name' => $table,
+            'url' => $foundUrl,
+            'last_check' => time(),
+            'link_type' => 'external',
+        ];
+        $this->brokenLinkRepository->addBrokenLink($link, false, [
+            'errorType' => 'exception',
+            'exception' => 'Do not link externally to ' . $forbiddenDomain,
+            'errno' => 42, ]);
+    }
+
+    private function findAllParsers(array $conf): iterable
+    {
+        return $this->softReferenceParserFactory->getParsersBySoftRefParserList($conf['softref'], ['subst']);
+    }
+
+    public function __invoke(BeforeRecordIsAnalyzedEvent $event): void
+    {
+        $table = $event->getTableName();
+        $forbiddenDomain = 'example.org';
+        if ($table !== 'tx_news_domain_model_news') {
+            return;
+        }
+        $results = $event->getResults();
+        $newsItem = $event->getRecord();
+        if (!str_contains((string)$newsItem['bodytext'], $forbiddenDomain)) {
```

Comment:
I don't agree. This field comes from the database and we have no guarantee about its format. if it was an integer it would fail str_contains.

Like this str_contains would correctly answer false. The only other solution is to test for the type first.